### ISSUE TITLE
Custom Password not being set due to JSON parse error

### ISSUE THREAD

**Issue by MarkusBauerBE** [CrunchyData/postgres-operator#3073]
Please ensure you do the following when reporting a bug:

- [x] Provide a concise description of what the bug is.
- [x] Provide information about your environment.
- [x] Provide clear steps to reproduce the bug.
- [x] Attach applicable logs. Please do not attach screenshots showing logs unless you are unable to copy and paste the log data.
- [x] Ensure any code / output examples are [properly formatted](https://docs.github.com/en/github/writing-on-github/basic-writing-and-formatting-syntax#quoting-code) for legibility.

Note that some logs needed to troubleshoot may be found in the `/pgdata/<CLUSTERNAME>/pg_log` directory on your Postgres instance.

An incomplete bug report can lead to delays in resolving the issue or the closing of a ticket, so please be as detailed as possible.

If you are looking for [general support](https://access.crunchydata.com/documentation/postgres-operator/latest/support/), please view the [support](https://access.crunchydata.com/documentation/postgres-operator/latest/support/) page for where you can ask questions.

Thanks for reporting the issue, we're looking forward to helping you!

## Overview

According to the documentation I should be able to update a custom users password by updating the "password" value in the users secret and setting the verifier to empty (according to this guide)[https://access.crunchydata.com/documentation/postgres-operator/v5/architecture/user-management/]. However the operator seems to run into an error parsing the secret.

## Environment

Please provide the following details:

- Platform: (`Kubernetes`,)
- Platform Version: (e.g. `1.21.2`, `5.0.4`)
- PGO Image Tag: (e.g. `ubi8-5.0.4-0`)
- Postgres Version (e.g. `14`)
- Storage: cinder-csi

### REPRO

Deploy a PG Cluster with a user (in this case gitlab)
Update the secret 
Notice that the cluster doesn't update the password

### EXPECTED

The cluster should be able to update the password

### ACTUAL

The cluster doesn't update the password (it tries but fails to parse the secret)

## Logs

```
time="2022-03-03T14:37:47Z" level=debug msg="wrote PostgreSQL users" file="internal/postgres/users.go:131" func=postgres.WriteUsersInPostgreSQL name=gitlab-postgres namespace=gitlab pod=gitlab-postgres-instance-8mzf-0 reconciler group=postgres-operator.crunchydata.com reconciler kind=PostgresCluster revision=b64656b8f stderr="psql:<stdin>:5: ERROR:  invalid input syntax for type json\nDETAIL:  Character with value 0x0a must be escaped.\nCONTEXT:  JSON data, line 1: ...s7k=:r5zTkH+4yC2P0nW8KfnSnbFBFMG1uGxHuZRxajmnpyg=\nCOPY input, line 1, column data: \"{\"databases\":[\"gitlab\"],\"options\":\"SUPERUSER CREATEDB\",\"username\":\"gitlab\",\"verifier\":\"SCRAM-SHA-256...\"\n" stdout= version=5.0.4-0 
```

The line it seems to yell at is the scram hash for the password.

```
apiVersion: v1    
data:
  password: <password>
  user: <user>
  dbname: Z2l0bGFi
  host: <gitlab host>
  jdbc-uri: <jdbc uri>
  port: NTQzMg==
  uri: <uri>
  verifier: ""
kind: Secret
metadata: 
  annotations:                                                                                                                                                    
    avp.kubernetes.io/path: {{ .Values.argo.vault.secretAnnotationPath | quote }}
  name: gitlab-postgres-pguser-gitlab
  namespace: gitlab
  labels: 
    postgres-operator.crunchydata.com/cluster: gitlab-postgres 
    postgres-operator.crunchydata.com/pguser: gitlab  
    postgres-operator.crunchydata.com/role: pguser  
type: Opaque
```

## Additional Information

My idea was to deploy the cluster but have the user secret managed by a hashicorp vault instance. Everything is deployed using ArgoCD and the ArgoCD vault plugin. Looking at the actually rendered secrets doesn't show any obvious JSON errors or anything and the secrets themselves can be edited and worked with just fine (K8s would yell if there was a JSON error in the secret itself).
Could this be an issue with the scram hash? It seems to use a lot of special characters

**Comment by MarkusBauerBE** [comment]
I think I may have stumbled upon the solution to my issue. It is a very peculiar one that seems to pop up in my daily business more and more: Base64 encoding using echo automatically appends a newline character at the end of the string you want to encode.

This seems to throw off some applications.

Basically what happened in my case was, as I was retrieving my password from Hashicorp Vault using the ArgoCD Vault plugin I had to provide placeholders for the aforementioned plugin to be able to find its relevant key in vault.
So my password is stored in a Vault secret behind the key "password" - in order for the plugin to find the value it needs to look like this: <password> but base64 encoded.

So what did I do? I entered:
```
echo "<password>" | base64
```
Which results in the following string: PHBhc3N3b3JkPgo=
Decoding this string again shows something that is easy to miss: A newline at the end. It is easy to miss because if you don't know that this can be an issue at all it'll just look like the command itself did the newline in order for your output to not look like:
```
<password>myuser@myhostwhatever
```
but rather like this:
```
<password>
myuser@myhostwhatever:
```

## How to avoid this
Use the -n flag for echo.
So:
```
echo -n "<password>" | base64
```
This results in: PHBhc3N3b3JkPg==

Note the slight difference between the encoded strings. It is subtle but it matters.

**tl;dr: Use the -n flag for echo when encoding K8s secret strings with base64**

Closing this as it was definitely my own fault.
